FirebaseAI/Sources/GenerateContentResponse.swift

```diff
@@ -299,6 +302,123 @@ public struct PromptFeedback: Sendable {
   }
 }
 
+/// Metadata returned to the client when grounding is enabled.
+///
+/// > Important: If using Grounding with Google Search, you are required to comply with the [Service
+/// Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with Google
+/// Search*.
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+public struct GroundingMetadata: Sendable {
+  /// A list of web search queries that the model performed to gather the grounding information.
+  /// These can be used to allow users to explore the search results themselves.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  public let webSearchQueries: [String]
+  /// A list of ``GroundingChunk`` structs. Each chunk represents a piece of retrieved content
+  /// (e.g., from a web page) that the model used to ground its response.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  public let groundingChunks: [GroundingChunk]
+  /// A list of ``GroundingSupport`` structs. Each object details how specific segments of the
+  /// model's response are supported by the `groundingChunks`.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  public let groundingSupports: [GroundingSupport]
+  /// Google search entry point for web searches.
+  /// This contains an HTML/CSS snippet that **must** be embedded in an app to display a Google
+  /// Search Entry point for follow-up web searches related to the model's "Grounded Response".
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  public let searchEntryPoint: SearchEntryPoint?
+
+  /// A struct representing the Google Search entry point.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  @available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+  public struct SearchEntryPoint: Sendable {
+    /// An HTML/CSS snippet that can be embedded in your app. The snippet is designed to avoid
+    /// undesired interaction with the rest of the page's CSS.
+    ///
+    /// To ensure proper rendering, it's recommended to display this content within a `WKWebView`.
+    public let renderedContent: String?
```

Comment:
> Do all of the new optional fields need to be optional?
-- @paulb777 

I think it's likely that `renderedContent` will always be non-nil when `GroundingMetadata.searchEntryPoint` is not `nil`. We _could_ consider making this non-optional.

Reply:
If we are re-evaluating mirroring the protos for this, we should make this non-optional.

Displaying a grounded result without displaying `renderedContent` is a violation of the service terms. To prevent this from happening, we should throw an error if `renderedContent` is nil. This approach should be extended to the struct `SearchEntrypoint`.

Thoughts? @andrewheard @paulb777

Reply:
changed in https://github.com/firebase/firebase-ios-sdk/pull/15014/commits/a63057aa3d2f5fd1833f35c70ce448bd457126b4